src/main/java/com/iexec/worker/pubsub/StompClient.java

```diff
@@ -0,0 +1,169 @@
+        this.stompClient.setMessageConverter(new MappingJackson2MessageConverter());
+        this.stompClient.setTaskScheduler(new ConcurrentTaskScheduler());
+        log.info("Created STOMP client");
+    }
+
+    /**
+     * Subscribe to a topic and provide a {@link StompFrameHandler}
+     * to handle received messages.
+     * 
+     * @param topic
+     * @param messageHandler an implementation of 
+     * @return
+     */
+    Subscription subscribeToTopic(String topic, StompFrameHandler messageHandler) {
+        return this.session.subscribe(topic, messageHandler);
+    }
+
+    @PostConstruct
+    private void init() {
+        requestNewSession();
+    }
+
+    /**
+     * Add new SessionRequestEvent to the queue. A queue listener
+     * will consume this event and create a new STOMP session.
+     * This does not raise an error if the queue is full.
+     */
+    private void requestNewSession() {
+        this.sessionRequestQueue.offer(new SessionRequestEvent());
+    }
+
+    /**
+     * Listen to new session request events and refresh the websocket
+     * connection by establishing a new STOMP session. Only one of 
+     * the received requests in a fixed time interval
+     * (SESSION_REFRESH_DELAY) will be processed. We use @Scheduled
+     * to start the watcher asynchronously with an initial delay and
+     * restart it in case a problem occurs.
+     * 
+     * @throws InterruptedException
+     */
+    @Scheduled(fixedDelay = 1000)
+    private void listenToSessionRequestEventsInTheQueue() throws InterruptedException {
+        while (true) {
+            // get the first request event or wait until available
+            this.sessionRequestQueue.take();
+            // wait some time for the wave of request events coming
+            // from possibly different threads to finish
+            TimeUnit.SECONDS.sleep(SESSION_REFRESH_DELAY);
+            log.info("Creating new STOMP session");
+            // purge redundant request events
+            this.sessionRequestQueue.clear();
+            // Only one attempt should pass through
+            log.debug("Sending new STOMP connection request");
+            this.stompClient.connect(webSocketServerUrl, new SessionHandler());
+        }
+    }
+
+    /**
+     * Provide callbacks to handle STOMP session establishment or
+     * failure.
+     */
+    private class SessionHandler extends StompSessionHandlerAdapter {
+
+        @Override
+        public void afterConnected(StompSession session, StompHeaders connectedHeaders) {
+            log.info("Connected to STOMP session [session: {}, isConnected: {}]",
+                    session.getSessionId(), session.isConnected());
+            StompClient.this.session = session;
+            // notify subscribers
+            eventPublisher.publishEvent(new SessionCreatedEvent());
+        }
+
+        @Override
+        public void handleException(StompSession session, @Nullable StompCommand command,
```

Comment:
Could you add a small comment explaining when this exception is triggered (same for handleTransportError)?

Reply:
Of course, yes!

Reply:
I added the original javadoc of the abstract methods in the interface `StompFrameHandler` [here](https://github.com/978740431/spring_readinig/blob/fcec314a82e167b61334a36289f5565bd5d23866/spring-messaging/src/main/java/org/springframework/messaging/simp/stomp/StompSessionHandler.java#L46)